src/compiler/checker.ts

```diff
@@ -10372,6 +10400,17 @@ namespace ts {
             return getUnionType([type1, type2]);
         }
 
+        function checkStringLiteralExpression(node: StringLiteral): Type {
+            // TODO (drosen): Do we want to apply the same approach to no-sub template literals?

```

Comment:
I'm choosing not to apply the same approach because we currently don't treat NoSubTemplates the same as strings in places like string indexing and overload resolution. I think it would be weird to treat them the same way here but not elsewhere.

For the record, I would prefer if we treated them the same elsewhere, but I don't want to conflate that with this PR.